src/storage/src/hummock/iterator/merge_inner.rs

```diff
@@ -12,73 +12,75 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::cmp::Ordering;
 use std::collections::binary_heap::PeekMut;
 use std::collections::{BinaryHeap, LinkedList};
 use std::sync::Arc;
 
 use async_trait::async_trait;
 use risingwave_hummock_sdk::VersionedComparator;
 
-use super::variants::*;
-use crate::hummock::iterator::{BoxedHummockIterator, HummockIterator};
+use crate::hummock::iterator::{
+    BoxedHummockIterator, DirectionEnum, HummockIterator, HummockIteratorDirection,
+};
 use crate::hummock::value::HummockValue;
 use crate::hummock::HummockResult;
 use crate::monitor::StateStoreMetrics;
 
-pub struct Node<'a, const DIRECTION: usize>(BoxedHummockIterator<'a>);
+pub struct Node<'a, D: HummockIteratorDirection, T> {
+    iter: BoxedHummockIterator<'a, D>,
+    extra_order_info: T,
```

Comment:
This field looks weird to me. My hunch is that we introduce extra complexity in MergeIterator implementation in order to reuse just a few common logics from UnorderedMergeIterator to implement OrderedMergeIterator. At the very least, I think we should separate the Node implementations for UnorderedMergeIterator and OrderedMergeIterator.

Reply:
Have separated the cmp implementation for two kinds of nodes.